packages/flutter_tools/lib/src/preview_device.dart

```diff
@@ -29,7 +29,7 @@ BundleBuilder _defaultBundleBuilder() {
   return BundleBuilder();
 }
 
-class PreviewDeviceDiscovery extends DeviceDiscovery {
+class PreviewDeviceDiscovery extends PollingDeviceDiscovery {
```

Comment:
I can't remember the reason this needed to be polling, but it might be worth adding a note in the comment if it's not obvious for future readers.